oak_runtime/src/node/http/tests.rs

```diff
@@ -197,35 +221,39 @@ fn create_communication_channel(runtime: &RuntimeProxy) -> (oak_abi::Handle, oak
     (init_receiver, invocation_receiver)
 }
 
-fn oak_node_simulator(runtime: &RuntimeProxy, invocation_receiver: oak_abi::Handle) {
-    // Get invocation message that contains the response_writer handle.
-    let invocation_receiver = Receiver::<HttpInvocation>::new(ReadHandle {
-        handle: invocation_receiver,
-    });
-    let invocation = invocation_receiver.receive(runtime).unwrap();
-    let resp = HttpResponse {
-        body: vec![],
-        status: http::status::StatusCode::OK.as_u16() as i32,
-        headers: hashmap! {},
-    };
-    invocation
-        .sender
-        .expect("Empty sender on invocation.")
-        .send(resp, runtime)
-        .unwrap();
+// Simulate an Oak node that responds to the specified number of requests. Since the Oak node blocks
+// until a new request is received, we need to know the number of requests that the test sends
+// beforehand. Alternatively, the node simulator could listen for incoming requests in an infinite
+// loop and have a termination mechanism that could be signalled from outside. However, specifying
+// the exact number of expected requests is more appropriate for testing purposes.
+fn oak_node_simulator(
+    number_of_requests: i32,
+    runtime: &RuntimeProxy,
+    invocation_receiver: oak_abi::Handle,
+) {
+    for _counter in 0..number_of_requests {
+        // Get invocation message that contains the response_writer handle.
+        let invocation_receiver = Receiver::<HttpInvocation>::new(ReadHandle {
+            handle: invocation_receiver,
+        });
```

Comment:
This seems like it should be created once and for all before the loop?

Reply:
Thanks. Done.